Failure of LinkHelperTest.testLinkHelperEditorActivation

Failed in [this build](https://github.com/eclipse-platform/eclipse.platform.ui/actions/runs/7029540124/job/19127333734?pr=1301) of #1301 

```java
org.eclipse.ui.tests.navigator.LinkHelperTest.testLinkHelperEditorActivation -- Time elapsed: 3.883 s <<< FAILURE!
java.lang.AssertionError: expected:<1> but was:<2>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at org.eclipse.ui.tests.navigator.LinkHelperTest.testLinkHelperEditorActivation(LinkHelperTest.java:97)
```

--- a/bundles/org.eclipse.ui.navigator/src/org/eclipse/ui/internal/navigator/actions/LinkEditorAction.java
+++ b/bundles/org.eclipse.ui.navigator/src/org/eclipse/ui/internal/navigator/actions/LinkEditorAction.java
@@ -149,9 +149,12 @@ public class LinkEditorAction extends Action implements ISelectionChangedListene
 
 			@Override
 			public void partBroughtToTop(IWorkbenchPart part) {
-				if (part instanceof IEditorPart && !ignoreEditorActivation) {
-					updateSelectionJob.schedule(NavigatorPlugin.LINK_HELPER_DELAY);
-				}
+				// Do not schedule a selection update here:
+				// 1. when the part is just brought to top (programmatically) but not activated,
+				// no link update is expected.
+				// 2. scheduling from both partActivated and partBroughtToTop can cause
+				// the job to run twice when the two events fire more than
+				// LINK_HELPER_DELAY apart (intermittent on slow machines).
 			}
 
 			@Override

